postgresql-async/src/main/java/com/github/jasync/sql/db/postgresql/encoders/PasswordEncoder.kt

```diff
@@ -0,0 +1,62 @@
+package com.github.jasync.sql.db.postgresql.encoders
+
+import com.github.jasync.sql.db.postgresql.messages.backend.ServerMessage
+import com.github.jasync.sql.db.postgresql.messages.frontend.ClientMessage
+import com.github.jasync.sql.db.postgresql.messages.frontend.PasswordMessage
+import com.github.jasync.sql.db.postgresql.messages.frontend.SASLInitialResponse
+import com.github.jasync.sql.db.postgresql.messages.frontend.SASLResponse
+import com.github.jasync.sql.db.postgresql.util.PasswordHelper
+import com.github.jasync.sql.db.util.ByteBufferUtils
+import com.github.jasync.sql.db.util.XXX
+import io.netty.buffer.ByteBuf
+import io.netty.buffer.Unpooled
+import java.nio.charset.Charset
+
+class PasswordEncoder(val charset: Charset) : Encoder {
```

Comment:
why did you rename that? it came out as a delete/add so hard to compare (and track history later)

Reply:
The intent was to reduce confusion. It's previous name was CredentialEncoder, but it really only handled passwords. Now that we have another authentication mechanism (SCRAM-SHA-256) that also deals with credentials, I thought it would be clearer to name distinguish this class from the other forms of authentication.